languages/ruby/lib/oso/polar/polar.rb

```diff
@@ -88,6 +88,49 @@ def self.join(separator, left, right)
         ffi_polar.validate_roles_config(validation_query_results)
       end
 
+      def get_allowed_resources(actor, action, klass) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+        resource = Variable.new 'resource'
+
+        # FIXME(gw) kind of a hack because of class autoreloading, maybe not needed?
+        if host.types.key? klass
+          class_name = host.types[klass].name
+        elsif host.types.key? klass.name
+          class_name = host.types[klass.name].name
+        else
+          raise "Unknown class `#{klass}`"
+        end
```

Comment:
Hmm this is kind of weird -- seems like we kind of want to store a mapping from user-supplied type name AND from class name so that you can look up the `UserType` based on a class even if the `name` they provided doesn't match.

```ruby
oso.register_class(Expense, "MyApp::Expense")
# ... later
oso.get_allowed_resources(actor, "read", Expense)
# OR
oso.get_allowed_resources(actor, "read", "Expense")
# OR
oso.get_allowed_resources(actor, "read", "MyApp::Expense")
```

Reply:
that's a good idea but what about name collisions?

Reply:
Probably going with the name of the registered class would be a safe bet? Not sure though. I think the most important test case to get working here would be:
```ruby
oso.register_class(Expense, "MyApp::Expense")

oso.get_allowed_resources(actor, "read", Expense)
```
Which should be able to find the `MyApp::Expense` name to send with the query. I think right now that would throw an `Unknown class "Expense"` error.

Reply:
Added a test. It works since we use the class object as a key as well as the name